lib/spack/spack/ci.py

```diff
@@ -2020,25 +2020,51 @@ def process_command(name, commands, repro_dir, run=True, exit_on_failure=True):
 
     Returns: the exit code from processing the command
     """
```

Comment:
That may need to change in general, but this PR was more targeted toward changing the `process_command` workflow, rather than the callers who use it. Ryan and I will likely refactor this in the larger CI support PR.

Reply:
I suppose then that whatever processes are using this PR right now, they don't happen to invoke `reproduce_ci_job`? I can see the value of updating the minimum set, but I think the functions which are not essential should have a `TODO` comment on them.

Reply:
Update: it has been pointed out to me that the goal of this PR is to refactor the underlying script invocation mechanism, but does not intend to make the CI commands runnable on Windows.